src/libpmemobj/heap.c

```diff
@@ -284,7 +285,7 @@ heap_bucket_acquire(struct palloc_heap *heap, uint8_t class_id,
 		b = arena->buckets[class_id];
 	} else {
 		b = (VEC_ARR(&heap->rt->arenas) \
-			[arena_id - 1])->buckets[class_id];
+			[IDX(arena_id)])->buckets[class_id];
```

Comment:
I was more thinking about macro (or function actually) for the whole expression: VEC_ARR(&heap->rt->arenas)[arena_id - 1]